### PR TITLE
Feat incremental build

### PR DESCRIPTION
**Pull request by kip-carr**
Adding ability to add in new images and not recalculate detect and match features.  Additional updates for not reading images and naming of feature files to include algorithm.
Added Docerfiles to help with building and moved imports to functions for lazy init so it doesnt complain if not using lightglue, etc.

### CODE REVIEW

File: opensfm/pairs_selection.py [hunk 1]
```diff
@@ -531,6 +535,8 @@ def match_candidates_by_time(
 
     pairs = set()
     for image_ref in images_ref:
+        if data.matches_exists(image_ref):
```
Comment (EliDavis3D) [inline opensfm/pairs_selection.py:538]: Have you tested this in a data set of 2 images.

1. Run match
2. Add 3rd image
3. Run match again
4. Observe changes to the matches files?

File: opensfm/pairs_selection.py [hunk 2]
```diff
@@ -558,6 +567,8 @@ def match_candidates_by_order(
 
     pairs = set()
     for i, image_ref in enumerate(images_ref):
+        if data.matches_exists(image_ref):
```
Comment (EliDavis3D) [inline opensfm/pairs_selection.py:570]: Have you tested this in a data set of 2 images.

    Run match
    Add 3rd image
    Run match again
    Observe changes to the matches files?

File: opensfm/features.py [hunk 3]
```diff
@@ -586,7 +577,9 @@ def extract_features_popsift(
 def extract_features_aliked(
         image: np.ndarray, config: Dict[str, Any], features_count: int
 ) -> Tuple[np.ndarray, np.ndarray]:
-    
+    import torch
```
[the remark this answers is not in the document]
Comment (jdcampbe) [reply opensfm/features.py:580]: Private fork might be an option, but I think when Alex went to fork it that wasn't available. You could only do a public fork.
I like the bda branch. That could be the stripped down version.

[the remark this answers is not in the document]
Comment (jdcampbe) [reply opensfm/features.py:580]: image

File: opensfm/features_processing.py [hunk 4]
```diff
@@ -236,16 +248,6 @@ def detect(
...
-    has_words = not need_words or data.words_exist(image)
-    has_features = data.features_exist(image)
-
-    if not force and has_features and has_words:
-        logger.info(
-            "Skip recomputing {} features for image {}".format(
-                data.feature_type().upper(), image
-            )
-        )
-        return
```
[the remark this answers is not in the document]
Comment (kip-carr) [reply opensfm/features_processing.py:248]: This got moved to the read image so it still always happens but it didnt need to be in both places and one function is called right after the other with no other references.

[the remark this answers is not in the document]
Comment (EliDavis3D) [reply opensfm/features_processing.py:248]: The reason I'd have it in both places is due to the multiple instance of OpenSFM writing to the same filesystem.

Both might get initialized, realize that pair needs to be computed. One does it before the other. The other then uses this code to detect that it now exists, and skips it

[the remark this answers is not in the document]
Comment (kip-carr) [reply opensfm/features_processing.py:248]: I think I get what you are saying but keeping this in both places wouldnt fix that issue.  If it gets to read and says it isnt there so keep the image and then moves to detect() and the file is there so it skips it we would have an issue b/c the compare would now not happen on the new image in this set against the new image in the other process.  I think you bring up a problem we will have to figure out with or without this check though.

**Review by EliDavis3D — APPROVED** [review]

**Pull request merged**